src/core-api/user/user.service.spec.ts

```
@@ -0,0 +1,158 @@
+        firstName: 'test',
+        lastName: 'case',
+        email: 'test@test.com',
+        phoneNumber: '+380961231212',
+      });
+
+      expect(actualResult).toEqual(user);
+    });
+
+    it('should throw BadRequestException', async () => {
+      userRepoSaveException = true;
+      const actualResult = userService.create({
+        firstName: 'test',
+        lastName: 'case',
+        email: 'test@test.com',
+        phoneNumber: '+380961231212',
+      });
+
+      await expect(actualResult).rejects.toThrowError(BadRequestException);
+    });
+  });
+
+  describe('update', () => {
+    it('should update the user', async () => {
+      const actualResult = await userService.update(1, {
+        lastName: 'jest',
+      });
+
+      expect(actualResult).toEqual(user);
+    });
+
+    it('should throw NotFoundException', async () => {
+      userRepoFindException = true;
+      const actualResult = userService.update(1, { lastName: 'jest' });
+
+      await expect(actualResult).rejects.toThrowError(NotFoundException);
+    });
+
+    it('should throw BadRequestException', async () => {
+      userRepoSaveException = true;
+      const actualResult = userService.update(1, { lastName: 'jest' });
+
+      await expect(actualResult).rejects.toThrowError(BadRequestException);
+    });
+  });
+
+  describe('get', () => {
+    it('should be called with correct parameters', async () => {
+      const filter = { limit: 50 } as Partial<GetUsersDto>;
+
+      await userService.get(filter);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(userRepository.find).toBeCalledWith({
+        skip: 0,
+        take: filter.limit,
+      });
+    });
+  });
+
+  describe('getOne', () => {
+    it('should return the user', async () => {
+      const actualResult = await userService.getOne(1);
+
+      expect(actualResult).toEqual(user);
+    });
+
+    it('should throw NotFoundException', async () => {
```

should throw exception if user not found